scripts/build-and-test.sh

```diff
@@ -19,10 +19,16 @@ function run_tests_with_gdb {
   for x in bin/**/*-tests; do scripts/run-with-gdb.sh "$x"; done
 }
 
+function run_tests {
+    for x in bin/**/*-tests; do "$x"; done
+}
+
+num_procs=$(lscpu -p | grep -v '^#' | sort -u -t, -k 2,4 | wc -l) # pysical cores
```

Comment:
```
++lscpu -p
++wc -l
++sort -u -t, -k 2,4
++grep -v '^#'
+num_procs=1
+build_beast
+/home/travis/boost_1_60_0/bjam toolset=gcc variant=asan address-model=64 -j1
```

Seems like Travis is just -j1 as per the usual implicit default. I'm using this in an updated branch:
```
if [[ `uname` == "Darwin" ]]; then
  num_jobs=`sysctl -n hw.ncpu`
else
  num_jobs=`nproc`
fi
```

This works on circle CI Docker and locally and on Travis:
```
+[[ Linux == \D\a\r\w\i\n ]]
++nproc
+num_jobs=2
```